frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts

```diff
@@ -140,186 +250,156 @@ export class ImportExperimentComponent implements OnInit {
-      enrollmentCompleteCondition: 'enum',
-      endOn: 'string',
-      revertTo: 'string',
-      tags: 'array',
-      logging: 'boolean',
-      group: 'string',
-      conditions: 'interface',
-      partitions: 'interface',
-      factors: 'interface',
-      queries: 'array',
-      stateTimeLogs: 'interface',
-      filterMode: 'string',
-      experimentSegmentInclusion: 'interface',
-      experimentSegmentExclusion: 'interface',
-      backendVersion: 'string',
-    };
-
-    const conditionSchema: Record<keyof ExperimentCondition, string> = {
-      id: 'string',
-      name: 'string',
-      description: 'string',
-      conditionCode: 'string',
-      assignmentWeight: 'number',
-      twoCharacterId: 'string',
-      order: 'number',
-      createdAt: 'string',
-      updatedAt: 'string',
-      versionNumber: 'number',
-      levelCombinationElements: 'interface',
-    };
-
-    const conditionSchemaForSimpleExp: Record<keyof ExperimentConditionForSimpleExp, string> = {
-      id: 'string',
-      name: 'string',
-      description: 'string',
-      conditionCode: 'string',
-      assignmentWeight: 'number',
-      twoCharacterId: 'string',
-      order: 'number',
-      createdAt: 'string',
-      updatedAt: 'string',
-      versionNumber: 'number',
-    };
-
-    const partitionSchema: Record<keyof ExperimentDecisionPoint, string> = {
-      id: 'string',
-      site: 'string',
-      target: 'string',
-      description: 'string',
-      twoCharacterId: 'string',
-      order: 'number',
-      createdAt: 'string',
-      updatedAt: 'string',
-      versionNumber: 'number',
-      excludeIfReached: 'boolean',
-    };
-
-    const factorSchema: Record<keyof ExperimentFactor, string> = {
-      name: 'string',
-      description: 'string',
-      order: 'number',
-      levels: 'interface',
-    };
-
-    const levelSchema: Record<keyof ExperimentLevel, string> = {
-      id: 'string',
-      name: 'string',
-      payload: 'string',
-      order: 'number',
-    };
-
-    const levelCombinationElementSchema: Record<keyof LevelCombinationElement, string> = {
-      id: 'string',
-      level: 'interface',
-    };
-
-    const missingProperties = this.checkForMissingProperties({ schema: experimentSchema, data: experiment });
-    const isFactorialExperiment = experiment.type === EXPERIMENT_TYPE.FACTORIAL;
-    let missingPropertiesFlag = true;
+  async validateExperimentJSON(experiment: Experiment) {
+    let missingProperties = this.checkForMissingProperties({ schema: this.experimentSchema, data: experiment });
+    const missingPropertiesList = missingProperties !== '' ? missingProperties.split(', ') : [];
+
+    if (missingPropertiesList.includes('backendVersion')) {
+      const currentBackendVersion = await this.versionService.getVersion();
+      experiment = { ...experiment, backendVersion: currentBackendVersion.toString() };
+      const index = missingPropertiesList.indexOf('filtebackendVersionrMode');
```

Comment:
`'filtebackendVersionrMode'` looks like a typo

Reply:
Oh yes, thanks for pointing this out. :)